tests/test_events.py

```diff
@@ -104,3 +151,131 @@ def test_publish(self):
             course=block.location.course,
             run=block.location.run,
         )
+
+    @data(*[
+        ['choice_0', 'an incorrect answer'],
+        ['choice_1', 'the correct answer'],
+        ['choice_2', 'a different incorrect answer'],
+    ])
+    @unpack
+    def test_problem(self, clicked_answer_id, expected_answer_text):
+        """
+        A problem should trigger an event which is captured
+        """
+        problem = self.get_problem()
+
+        problem.handle_ajax('problem_check', {
+            "input_i4x-SGAU-SGA101-problem-"
+            "2582bbb68672426297e525b49a383eb8_2_1": clicked_answer_id
+        })
+        assert RapidResponseSubmission.objects.count() == 1
+        obj = RapidResponseSubmission.objects.first()
+        assert obj.user_id == self.instructor.id
+        assert obj.course_id == self.course.course_id
+        assert obj.problem_id.map_into_course(
+            self.course.course_id
+        ) == problem.location
+        assert obj.answer_text == expected_answer_text
+        assert obj.answer_id == clicked_answer_id
+
+    def test_double_click(self):
+        """
+        Only the last click should get captured
+        """
+        problem = self.get_problem()
+
+        for answer in ('choice_0', 'choice_1', 'choice_2'):
+            problem.handle_ajax('problem_check', {
+                "input_i4x-SGAU-SGA101-problem-"
+                "2582bbb68672426297e525b49a383eb8_2_1": answer
+            })
+
+        assert RapidResponseSubmission.objects.count() == 1
+        obj = RapidResponseSubmission.objects.first()
+        assert obj.user_id == self.instructor.id
+        assert obj.course_id == self.course.course_id
+        assert obj.problem_id.map_into_course(
+            self.course.course_id
+        ) == problem.location
+        # Answer is the first one clicked
+        assert obj.answer_text == 'a different incorrect answer'
+        assert obj.answer_id == 'choice_2'  # the last one picked
+
+    def assert_event_parsing(self, modification_func, success):
+        """
+        Assert what happens when the event is parsed
+        """
+        with open(os.path.join(BASE_DIR, "..", "test_data", "example_event.json")) as f:
+            example_event = json.load(f)
+        modification_func(example_event)
+        SubmissionRecorder().send(example_event)
+        if success:
+            assert RapidResponseSubmission.objects.count() == 1
+            obj = RapidResponseSubmission.objects.first()
+            assert obj.user_id == example_event['context']['user_id']
+            assert obj.problem_id == UsageKey.from_string(
+                example_event['event']['problem_id']
+            )
+            # Answer is the first one clicked
+            assert obj.answer_text == 'an incorrect answer'
+            assert obj.answer_id == 'choice_0'
+            assert obj.event == example_event
+        else:
+            assert RapidResponseSubmission.objects.count() == 0
```

Comment:
The test cases below can be made easier to read and reason about. Pretty sure you can use pytest fixtures, so something like this might work:

```python
@pytest.fixture()
def example_event():
    with open(os.path.join(BASE_DIR, "..", "test_data", "example_event.json")) as f:
        yield json.load(f)

class TestEvents(ModuleStoreTestCase):
    #...
    def assert_successful_event_parsing(self, example_event_data):
        assert RapidResponseSubmission.objects.count() == 1
        # ....

    def assert_unsuccessful_event_parsing(self):
        assert RapidResponseSubmission.objects.count() == 0

    def test_example_event(self, example_event):
        SubmissionRecorder().send(example_event)
        self.assert_successful_event_parsing(example_event)

    def test_missing_user(self, example_event):
        del example_event['context']['user_id']
        SubmissionRecorder().send(example_event)
        self.assert_unsuccessful_event_parsing()
```

Reply:
You can use fixtures but they have to set the instance field, it's not allowed to pass via parameter, see the boxes at the bottom here: https://docs.pytest.org/en/latest/unittest.html#mixing-fixtures